### ISSUE TITLE
test_slow_wrapper fails on pypy3

### ISSUE THREAD

**Issue by mgorny** [micheles/decorator#124]
```
test_slow_wrapper (__main__.ExtraTestCase) ... calling __setitem__ with args (defaultdict(<class 'list'>, {}), 'x', [1]), {}
FAIL
[...]

======================================================================
FAIL: test_slow_wrapper (__main__.ExtraTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "src/tests/test.py", line 193, in test_slow_wrapper
    doc.trace(defaultdict.__getitem__)
AssertionError: ValueError not raised

----------------------------------------------------------------------
Ran 24 tests in 6.027s

FAILED (failures=1)
```

```
$ pypy3 --version
Python 3.7.10 (cd00c77c619f02c43c23e6a2514e5b685230ef02, May 14 2021, 20:08:22)
[PyPy 7.3.5 with GCC 11.1.0]
```

(that's 7.3.5rc2 actually)

**Comment by micheles** [comment]
That test is testing an implementation detail of CPython and I should change it.

**Comment by micheles** [comment]
This is fixed in release 5.0.9.

**Comment by mgorny** [comment]
Thanks for the fix! I can confirm that 5.0.9 passes all tests for me.
